MudBaseInput: Add "ShrinkLabel" parameter that prevents label from moving

Etzix opened the pull request:
<!--
MAKE SURE TO READ THE CONTRIBUTING GUIDE BEFORE CREATING A PR
https://github.com/MudBlazor/MudBlazor/blob/dev/CONTRIBUTING.md

Testing sections can be removed for documentation changes
-->

<!-- Provide a general summary of your changes in the Title above -->
<!-- Keep the title short and descriptive, as it will be used as a commit message -->


## Description
<!-- Describe your changes in detail any why -->
<!-- Note any issues that are resolved by this PR -->
<!-- e.g. resolves #1337 or fixes #9310 -->
Adds support for a "Shrink" parameter to MudTextField, MudAutocomplete and MudSelect. The Shrink parameter prevents the label from moving down into the field when the field is empty. This is inspired by the prop of the same name in Material UI https://mui.com/material-ui/api/input-label/#input-label-prop-shrink

This **partially** fixes #1114

## How Has This Been Tested?
<!-- All PR's should implement unit tests if possible -->
<!-- Please describe how you tested your changes. -->
<!-- Have you created new tests or updated existing ones? -->
<!-- e.g. unit | visually | none -->
This has been tested visually.

## Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

<!-- If you made any visual changes, provide screenshots of before/after, it its moving parts, please provide high quality gif, wemb or mp4 -->

![image](https://github.com/MudBlazor/MudBlazor/assets/31126885/6b77ea09-d69b-4f7e-b3d2-164dbce3cab3)


## Checklist:
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] The PR is submitted to the correct branch (`dev`).
- [x] My code follows the code style of this project.
- [ ] I've added relevant tests.

Commit:
MudBaseInput: Added "Shrink" parameter that prevents label from moving.

Shrink support in:
* MudTextField
* MudAutocomplete
* MudSelect

henon reviewed (changes requested):
LGTM with respect to the code changes. Would be nice to add to the docs so this new parameter is represented there.  

For instance, like the documentation of the Dense parameter:
![image](https://github.com/MudBlazor/MudBlazor/assets/44090/27051acd-5790-4ab1-ac69-92819c238516)

Commit:
MudBaseInput: Changed name to ShrinkLabel and added example to docs.

Etzix commented:
> LGTM with respect to the code changes. Would be nice to add to the docs so this new parameter is represented there.


@henon I added an example in TextFieldPage and changed the name of the property to "ShrinkLabel" to make it clear that its the label we are shrinking, not the input.

henon commented:
Thanks @Etzix

headauto commented:
is this working? going through the playground it doesn't seem to be working:
![image](https://github.com/MudBlazor/MudBlazor/assets/102734032/c0108e8b-0d0b-422a-b14f-63e28196cfee)

Adding the property ShrinkLabel doesn't seem to apply the mud-shrink class:
![image](https://github.com/MudBlazor/MudBlazor/assets/102734032/40b71126-34ca-492b-8817-63d74fd7db4c)

Sorry if I'm commenting in the wrong place.

mikes-gh commented:
@headauto Due to a deploy issue (now fixed) the try.mudblazor.com was on the previous 6.15.0 version.
Now updated to 6.16.0 It should work.

luis-fss commented:
It would be so nice to have a global setting for this instead of manually putting it everywhere...

henon commented:
That could still be done. We have `MudGlobal` for that. The inputs could take their default value from it.
